helsingborg-stad/app-mitt-helsingborg
# 🐛  bug: fix handling of multiple forms with same type

## Explain the changes you’ve made
Fixed a bug where the application only took the first occurrence of a form with a specific type instead of multiple.

## Explain why these changes are made
The application fetches multiples forms from the backend and stores them. Afterwards the application sorts out a form with a specific type, ignoring if multiple forms with the same type exist. This means if we have multiple forms with the same type, we might sort out forms that doesn't exist in a case attached to a specific user, which in turn means that this case will never be shown.

## Explain your solution
Before, we used javascript **find** for sorting out forms for a specific type, but find stop its execution when it finds the first occurrence that matches a condition. This was now changed using javascript **filter** instead which returns an array.
 
## How to test
1. Checkout this branch
2. Fire upp the simulator by running the command `yarn ios`
3. Make sure you have at least two forms with the same type
4. Make sure you have a case with two forms (forms created in step 3)
5. Make sure that one of the above form is added to `currentFormId` in the case
6. Check the CaseOverview page and check if the currentFormId-case is shown

## Tested environments

- [] Storybook on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.
